Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp

```diff
@@ -77,10 +81,20 @@ void EnhancementsWidget::CreateWidgets()
       std::max(Config::Get(Config::GFX_EFB_SCALE), Config::Get(Config::GFX_MAX_EFB_SCALE));
   for (int scale = static_cast<int>(resolution_options.size()); scale <= max_efb_scale; scale++)
   {
-    resolution_options.append(tr("%1x Native (%2x%3)")
-                                  .arg(QString::number(scale),
-                                       QString::number(static_cast<int>(EFB_WIDTH) * scale),
-                                       QString::number(static_cast<int>(EFB_HEIGHT) * scale)));
+    const QString scale_text = QString::number(scale);
+    const QString width_text = QString::number(static_cast<int>(EFB_WIDTH) * scale);
+    const QString height_text = QString::number(static_cast<int>(EFB_HEIGHT) * scale);
+    const int extra_index = resolution_options.size() - 2;
+    const QString& extra_text = resolution_extra_options[extra_index];
```

Comment:
This is out-of-bounds if the user hase configured `GFX_MAX_EFB_SCALE` to something higher than 12.

Reply:
I have no idea what happened, code seems to make no sense right now 😅.

Reply:
Oh it's because I'm growing 'resolution_extra_options' every iteration. So it starts from index zero and then 1 2...

Reply:
But it can indeed go out of bounds. Will fix.